util/geo.py

```diff
@@ -0,0 +1,188 @@
+            round(self.longitude, 6) == round(other.longitude, 6)
+        ):
+            return True
+        else:
+            return False
+
+    @classmethod
+    def normalize_location_input(cls, location):
+        """
+        Convert any of several formats to a 3-tuple of (latitude, longitude, srid), where
+            * latitude and longitude are floats
+            * srid is an integer or None
+
+        location may be any one of:
+
+            - A 2-tuple of (latitude, longitude)
+            - A 3-tuple of (latitude, longitude, srid)
+            - A comma and/or space separated string with 'latitude, longitude'
+            - A Well Known Text string of type Point, such as 'POINT(longitude latitude)'
+            - An Extended WKT string of type Point, such as 'SRID=4326;POINT(longitude latitude)'
+        """
+        (latitude, longitude, srid) = (None, None, None)
+
+        if isinstance(location, tuple):
+            if len(location) == 2:
+                (latitude, longitude) = location
+            elif len(location) == 3:
+                (latitude, longitude, srid) = location
+        elif isinstance(location, str):
+            if ',' in location or (' ' in location and not location.upper().startswith(('POINT', 'SRID'))):
+                # If it's got a comma, or doesn't start with POINT or SRID, it's not a WKT/EWKT Point string
+                match = LATLONG_STRING_REGEX.match(location)
+                if match:
+                    latitude = match.group('latitude')
+                    longitude = match.group('longitude')
+            elif location.upper().startswith(('POINT', 'SRID')):  # it's a WKT/EWKT string
+                match = LATLONG_WKT_EWKT_REGEX.match(location)
+                if match:
+                    srid = match.groupdict()['srid']
+                    latitude = match.group('latitude')
+                    longitude = match.group('longitude')
+
+        # Perform type coercion on the values we got
+        try:
+            latitude = float(latitude)
+            assert abs(latitude) <= 90.0
+            longitude = float(longitude)
+            assert abs(longitude) <= 180.0
+        except (TypeError, ValueError, AssertionError):
+            (latitude, longitude) = (None, None)  # These live or die together. No sense returning only one.
+
+        try:
+            assert (latitude and longitude)       # If lat/long didn't pass, srid doesn't matter
+            assert str(srid) == str(int(srid))    # srid has to be a real int, even if it's in a string
+            srid = int(srid)
+        except (TypeError, ValueError, AssertionError):
+            srid = None
+
+        # At this point either we've got real values from the tuple or string, or the location
+        # passed in wasn't a string or a tuple, so we can safely ignore it and return nothing.
+        return (latitude, longitude, srid)
+
+    @classmethod
+    def location_in_ocean(cls, location):
```

Comment:
Do you plan to make use of this outside of tests?

Reply:
I was going to performance test it and use it if it's a cheap operation, which I think it should be. I just wanted some very simple protection against having to do the more expensive geographic operations on obviously unusable locations. Possibly a premature / unnecessary optimization, can take it out if it seems silly.